scripts/viewer/run_viewer.py

```diff
@@ -61,9 +61,18 @@ def main(self) -> None:
     def _start_viewer(self, config: TrainerConfig, pipeline: Pipeline):
         base_dir = config.get_base_dir()
         viewer_log_path = base_dir / config.viewer.relative_log_filename
-        viewer_state, banner_messages = viewer_utils.setup_viewer(
-            config.viewer, log_filename=viewer_log_path, datapath=pipeline.datamanager.get_datapath()
-        )
+
+        # NOTE: for generative pipeline(s), set the data path on-the-fly.
+        if pipeline.config.generative:
+            assert pipeline.datamanager.get_datapath() is None
+            CONSOLE.print(f"Generative models should not have data path. Setting it to {str(base_dir)}")
+            viewer_state, banner_messages = viewer_utils.setup_viewer(
+                config.viewer, log_filename=viewer_log_path, datapath=base_dir
+            )
+        else:
+            viewer_state, banner_messages = viewer_utils.setup_viewer(
+                config.viewer, log_filename=viewer_log_path, datapath=pipeline.datamanager.get_datapath()
```

Comment:
I could potentially imagine generative methods that do rely on underlying data and therefore have a datapath. Maybe we should just default to `base_dir` if `pipeline.datamanager.get_datapath()` is `None`

Reply:
Thank you for your comment.

Agree, a conditional or editing setup may expect data that could be used as a starting point.